src/game/Tactical/Arms_Dealer_Init.cc

```diff
@@ -1015,27 +1017,23 @@ BOOLEAN CanDealerRepairItem(ArmsDealerID const ubArmsDealer, UINT16 const usItem
 		return(FALSE);
 	}
 
-	switch ( ubArmsDealer )
+	auto dealer = GetDealer(ubArmsDealer);
+	if (dealer->type == ArmsDealerType::ARMS_DEALER_REPAIRS)
 	{
-		case ARMS_DEALER_ARNIE:
-		case ARMS_DEALER_PERKO:
+		if (!dealer->hasFlag(ArmsDealerFlag::REPAIRS_ELECTRONICS))
```

Comment:
It'd be cleaner without negation, since both cases are covered, but it's also just a minor nit.